### ISSUE TITLE
Clarify how to use bundles correctly

### ISSUE THREAD

**Issue by smac89** [systemjs/systemjs#1699]
From my understanding after reading the docs, I should be able to specify something like

```js
bundles: {
    'polyfills.js': ['main.js']
}
```

So that when I do `System.import ('main.js')`, `polyfills.js` is imported before `main.js`. Is this right?
I am looking for a way to inform `SystemJS` that `polyfills` needs to be imported first before `main.js`. Right now, I have just a single import to `main.js`, but I get a lot of errors because the things that are in `polyfills.js` have not been imported.

I also tried:
```js
meta: {
    'main.js': {
        deps: ['polyfills.js'],
        format: 'global'
    }
}
```
And

```js
depCache: {
    'main.js': ['polyfills.js']
}
```

But no luck. What am I missing?

**Comment by SerkanSipahi** [comment]
There is not a documentary about using the fileExtension (.js) in your bundle config:
```js
bundles: {
    'polyfills.js': ['main.js']
}
```
see: https://github.com/systemjs/systemjs/blob/master/docs/config-api.md#bundles

That would be correct:
```js
bundles: {
    'polyfills.': ['main']
}
```
and in packages you can define the file-extension:
```js
SystemJS.config({
    packages: {
        "src": {
            "defaultExtension": "js"
        },
        "test": {
            "defaultExtension": "js"
        },
        "node_modules": {
            "defaultExtension": "js"
        }
    }
});
see: https://github.com/systemjs/systemjs/blob/master/docs/config-api.md#packages

```

**Comment by smac89** [comment]
@SerkanSipahi 

Thanks! I finally got it working, but I did need to have the extension in there to make it work. I think the problem is that systemjs builder does not include these bundles when I build a static bundle, so I will raise the issue there.

Just to clarify, this is what worked for me:

```js
bundles: {
    'polyfills.js': ['build/main.js'] // main.js is in the build folder
}

packages: {
   'build': {
       main: 'main',
       defaultExtension: 'js'
   }
}
```

Even though I specify the `defaultExtension: 'js'` in the build package, SystemJS would not import `polyfills.js' if the `build/main.js` did not have that `.js` extension to it. Not sure if this is a bug, but that was my experience. I'm also using the latest version of systemjs (version `0.20.14`) if that helps,
